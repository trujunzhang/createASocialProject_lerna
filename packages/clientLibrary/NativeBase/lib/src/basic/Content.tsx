import * as React from 'react'
import * as ReactNative from 'react-native'
import * as PropTypes from 'prop-types'
import { KeyboardAwareScrollView } from '@app/react-native-keyboard-aware-scroll-view'
import { connectStyle } from '@app/native-base-shoutem-theme'
import mapPropsToStyleNames from '../utils/mapPropsToStyleNames'
import { platformVariables as variable } from '@app/native-base-variables'
import { isIphoneX } from '@app/react-native-iphone-x-helper'

export interface IContentProps {
  style?: ReactNative.ViewStyle | Array<ReactNative.ViewStyle> | any
  /**
     * The theme prop can be applied to any component of NativeBase.
     */
  refreshing?: boolean
  refreshControl?: object | any
  theme?: Object
  padder?: boolean
  disableKBDismissScroll?: boolean | any
  enableResetScrollToCoords?: boolean
  contentOffset?: Object | any
  scrollEnabled?: boolean
  contentContainerStyle?: ReactNative.ViewStyle | Array<ReactNative.ViewStyle>
  keyboardShouldPersistTaps?: string | any
  keyboardDismissMode?: string | any
}
class Content extends React.Component<IContentProps, any> {
  private _root: any
  private _scrollview: any
  static contextTypes = {
    theme: PropTypes.object
  }
  constructor(props) {
    super(props)
    this.state = {
      orientation: 'portrait'
    }
  }
  layoutChange(val) {
    let maxComp = Math.max(variable.deviceWidth, variable.deviceHeight)
    if (val.width >= maxComp) this.setState({ orientation: 'landscape' })
    else {
      this.setState({ orientation: 'portrait' })
    }
  }

  calculateLeft(mode, inSet) {
    let inset: any = null
    if (inSet !== undefined) {
      inset = inSet
    } else {
      inset = variable.Inset
    }
    const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape
    let leftPadder = null
    if (this.props.style[1] !== undefined) {
      if (
        this.props.style[1].padding !== undefined ||
        this.props.style[1].paddingLeft !== undefined
      ) {
        leftPadder =
          (this.props.style[1].paddingLeft
            ? this.props.style[1].paddingLeft
            : this.props.style[1].padding) + InsetValues.leftInset
      }
    } else if (
      this.props.style.padding !== undefined ||
      this.props.style.paddingLeft !== undefined
    ) {
      leftPadder =
        (this.props.style.paddingLeft ? this.props.style.paddingLeft : this.props.style.padding) +
        InsetValues.leftInset
    } else {
      leftPadder = InsetValues.leftInset
    }
    return leftPadder
  }
  calculateRight(mode, inSet) {
    let inset: any = null
    if (inSet !== undefined) {
      inset = inSet
    } else {
      inset = variable.Inset
    }
    const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape
    let rightPadder = null
    if (this.props.style[1] !== undefined) {
      if (
        this.props.style[1].padding !== undefined ||
        this.props.style[1].paddingRight !== undefined
      ) {
        rightPadder =
          (this.props.style[1].paddingRight
            ? this.props.style[1].paddingRight
            : this.props.style[1].padding) + InsetValues.rightInset
      }
    } else if (
      this.props.style.padding !== undefined ||
      this.props.style.paddingRight !== undefined
    ) {
      rightPadder =
        (this.props.style.paddingRight ? this.props.style.paddingRight : this.props.style.padding) +
        InsetValues.rightInset
    } else {
      rightPadder = InsetValues.rightInset
    }
    return rightPadder
  }
  render() {
    const variables = this.context.theme
      ? this.context.theme['@@shoutem.theme/themeStyle'].variables
      : variable
    return isIphoneX() ? (
      <KeyboardAwareScrollView
        automaticallyAdjustContentInsets={false}
        resetScrollToCoords={(this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }) as any}
        keyboardShouldPersistTaps={
          this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled'
        }
        ref={(c) => {
          this._scrollview = c
          this._root = c
        }}
        {...this.props}
        onLayout={(e) => this.layoutChange(e.nativeEvent.layout)}
        style={[
          this.props.style,
          {
            paddingLeft: this.calculateLeft(this.state.orientation, variables.Inset)
          }
        ]}
        contentContainerStyle={[
          { padding: this.props.padder ? variables.contentPadding : undefined },
          this.props.contentContainerStyle
        ]}>
        {this.props.children}
      </KeyboardAwareScrollView>
    ) : (
        <KeyboardAwareScrollView
          automaticallyAdjustContentInsets={false}
          resetScrollToCoords={(this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }) as any}
          keyboardShouldPersistTaps={
            this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled'
          }
          ref={(c) => {
            this._scrollview = c
            this._root = c
          }}
          {...this.props}
          contentContainerStyle={[
            { padding: this.props.padder ? variables.contentPadding : undefined },
            this.props.contentContainerStyle
          ]}>
          {this.props.children}
        </KeyboardAwareScrollView>
      )
  }
}

// Content.propTypes = {
//     style: PropTypes.oneOfType([
//         PropTypes.object,
//         PropTypes.number,
//         PropTypes.array
//     ]),
//     padder: PropTypes.bool,
//     disableKBDismissScroll: PropTypes.bool,
//     enableResetScrollToCoords: PropTypes.bool,
//     keyboardShouldPersistTaps: PropTypes.string
// };

const StyledContent = connectStyle('NativeBase.Content', {}, mapPropsToStyleNames)(Content)

export { StyledContent as Content }
