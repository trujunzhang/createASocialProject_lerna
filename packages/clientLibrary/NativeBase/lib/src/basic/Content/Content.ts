import * as React from 'react'
import { ReactNative, RnTextStyleProp, RnViewStyleProp } from '@appUtils/rnCommonProps'

/**
 * see Widget Content.js
 */
export interface ContentProps {
  /**
   * The theme prop can be applied to any component of NativeBase.
   */
  refreshing?: boolean
  refreshControl?: object
  theme?: Object
  padder?: boolean
  disableKBDismissScroll?: boolean
  enableResetScrollToCoords?: boolean
  contentOffset?: Object
  scrollEnabled?: boolean
  style?: RnViewStyleProp | Array<RnViewStyleProp>
  contentContainerStyle?: RnViewStyleProp | Array<RnViewStyleProp>
  keyboardShouldPersistTaps?: string
  keyboardDismissMode?: string
}
