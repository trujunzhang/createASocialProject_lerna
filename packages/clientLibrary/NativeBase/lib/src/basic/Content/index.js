import * as React from 'react';
import * as PropTypes from 'prop-types';
// TODO:DJZHANG(NB)
import { getKeyboardAwareScrollView } from '@appUtils/cacheHelper';
import { isIphoneX } from 'react-primitives';
// import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import { connectStyle } from 'native-base-shoutem-theme';
import { platformVariables as variable } from '@appVariables/theme/variables/platform'; // variables // [variable]
import { mapPropsToStyleNames } from '@app/tools';
class Content extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            orientation: 'portrait'
        };
    }
    layoutChange(val) {
        let maxComp = Math.max(variable.deviceWidth, variable.deviceHeight);
        if (val.width >= maxComp)
            this.setState({ orientation: 'landscape' });
        else {
            this.setState({ orientation: 'portrait' });
        }
    }
    calculateLeft(mode, inSet) {
        let inset = null;
        if (inSet !== undefined) {
            inset = inSet;
        }
        else {
            inset = variable.Inset;
        }
        const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape;
        let leftPadder = null;
        if (this.props.style[1] !== undefined) {
            if (this.props.style[1].padding !== undefined ||
                this.props.style[1].paddingLeft !== undefined) {
                leftPadder =
                    (this.props.style[1].paddingLeft
                        ? this.props.style[1].paddingLeft
                        : this.props.style[1].padding) + InsetValues.leftInset;
            }
        }
        else if (this.props.style.padding !== undefined ||
            this.props.style.paddingLeft !== undefined) {
            leftPadder =
                (this.props.style.paddingLeft ? this.props.style.paddingLeft : this.props.style.padding) +
                    InsetValues.leftInset;
        }
        else {
            leftPadder = InsetValues.leftInset;
        }
        return leftPadder;
    }
    calculateRight(mode, inSet) {
        let inset = null;
        if (inSet !== undefined) {
            inset = inSet;
        }
        else {
            inset = variable.Inset;
        }
        const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape;
        let rightPadder = null;
        if (this.props.style[1] !== undefined) {
            if (this.props.style[1].padding !== undefined ||
                this.props.style[1].paddingRight !== undefined) {
                rightPadder =
                    (this.props.style[1].paddingRight
                        ? this.props.style[1].paddingRight
                        : this.props.style[1].padding) + InsetValues.rightInset;
            }
        }
        else if (this.props.style.padding !== undefined ||
            this.props.style.paddingRight !== undefined) {
            rightPadder =
                (this.props.style.paddingRight ? this.props.style.paddingRight : this.props.style.padding) +
                    InsetValues.rightInset;
        }
        else {
            rightPadder = InsetValues.rightInset;
        }
        return rightPadder;
    }
    render() {
        const variables = this.context.theme
            ? this.context.theme['@@shoutem.theme/themeStyle'].variables
            : variable;
        const KeyboardAwareScrollView = getKeyboardAwareScrollView();
        return isIphoneX() ? (React.createElement(KeyboardAwareScrollView, Object.assign({ name: 'keyboardAwareScrollView', automaticallyAdjustContentInsets: false, resetScrollToCoords: this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }, keyboardShouldPersistTaps: this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled', ref: (c) => {
                this._scrollview = c;
                this._root = c;
            } }, this.props, { onLayout: (e) => this.layoutChange(e.nativeEvent.layout), style: [
                this.props.style,
                {
                    paddingLeft: this.calculateLeft(this.state.orientation, variables.Inset)
                }
            ], contentContainerStyle: [
                { padding: this.props.padder ? variables.contentPadding : undefined },
                this.props.contentContainerStyle
            ] }), this.props.children)) : (React.createElement(KeyboardAwareScrollView, Object.assign({ name: 'keyboardAwareScrollView', automaticallyAdjustContentInsets: false, resetScrollToCoords: this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }, keyboardShouldPersistTaps: this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled', ref: (c) => {
                this._scrollview = c;
                this._root = c;
            } }, this.props, { contentContainerStyle: [
                { padding: this.props.padder ? variables.contentPadding : undefined },
                this.props.contentContainerStyle
            ] }), this.props.children));
    }
}
Content.contextTypes = {
    theme: PropTypes.object
};
// Content.propTypes = {
//   style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
//   padder: PropTypes.bool,
//   disableKBDismissScroll: PropTypes.bool,
//   enableResetScrollToCoords: PropTypes.bool,
//   keyboardShouldPersistTaps: PropTypes.string
// }
const StyledContent = connectStyle('NativeBase.Content', {}, mapPropsToStyleNames)(Content);
const TabContent = StyledContent;
export { StyledContent as Content, TabContent };
//# sourceMappingURL=index.js.map