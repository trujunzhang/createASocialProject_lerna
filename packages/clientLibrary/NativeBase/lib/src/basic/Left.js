import React, { Component } from "react";
import PropTypes from "prop-types";
import { View, ViewPropTypes } from "react-primitives";

import { connectStyle } from "@app/native-base-shoutem-theme";
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Left extends Component {
  render() {
    // debugger
    return <View ref={c => (this._root = c)} {...this.props} />;
  }
}

Left.propTypes = {
  ...ViewPropTypes,
  style: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.number,
    PropTypes.array
  ])
};

const StyledLeft = connectStyle("NativeBase.Left", {}, mapPropsToStyleNames)(
  Left
);

export { StyledLeft as Left };
