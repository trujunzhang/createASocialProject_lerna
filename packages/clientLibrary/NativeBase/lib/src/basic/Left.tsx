import * as React from 'react'
import * as ReactNative from 'react-native'
import * as PropTypes from 'prop-types'
import { View, ViewProps } from 'react-primitives'

import { connectStyle } from '@app/native-base-shoutem-theme'
import mapPropsToStyleNames from '../utils/mapPropsToStyleNames'

export interface ILeftProps extends ViewProps {
  style?: ReactNative.ViewStyle | Array<ReactNative.ViewStyle>
}
class Left extends React.Component<ILeftProps, any> {
  private _root: any

  render() {
    return <View ref={(c) => (this._root = c)} {...this.props} />
  }
}

// Left.propTypes = {
// ...ViewPropTypes,
// style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array])
// }

const StyledLeft = connectStyle('NativeBase.Left', {}, mapPropsToStyleNames)(Left)

export { StyledLeft as Left }
