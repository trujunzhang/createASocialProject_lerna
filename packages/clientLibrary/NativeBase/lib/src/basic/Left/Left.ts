import * as React from 'react'

import { ReactNative, RnViewStyleProp } from '@appUtils/rnCommonProps'

export interface LeftProps {
  style?: RnViewStyleProp | Array<RnViewStyleProp>
}
