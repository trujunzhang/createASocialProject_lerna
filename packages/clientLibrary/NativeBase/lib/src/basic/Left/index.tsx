import * as React from 'react'
import * as PropTypes from 'prop-types'
import { View } from 'react-primitives'

import { connectStyle } from 'native-base-shoutem-theme'
import { mapPropsToStyleNames } from '@app/tools'

class Left extends React.Component<any, any> {
  private _root: any
  render() {
    return <View name={'left'} ref={(c) => (this._root = c)} {...this.props} />
  }
}

// Left.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array])
// }

const StyledLeft = connectStyle('NativeBase.Left', {}, mapPropsToStyleNames)(Left)

export { StyledLeft as Left }
