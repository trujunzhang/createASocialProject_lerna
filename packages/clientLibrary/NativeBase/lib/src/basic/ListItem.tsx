import * as React from 'react'
import * as ReactNative from 'react-native';
import {
  TouchableHighlight,
  Platform,
  TouchableNativeFeedback,
  View
} from "react-primitives";

import { connectStyle } from "@app/native-base-shoutem-theme";
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";
import { platformVariables as variable } from '@app/native-base-variables'

export interface IListItemProps extends TouchableHighlight {
  style?: ReactNative.ViewStyle | Array<ReactNative.ViewStyle>
}
class ListItem extends React.Component<IListItemProps, any> {
  private _root: any
  static contextTypes = {
    theme: PropTypes.object
  };
  render() {
    const variables = this.context.theme
      ? this.context.theme["@@shoutem.theme/themeStyle"].variables
      : variable;

    if (
      Platform.OS === "ios" ||
      Platform.OS === "web" ||
      variables.androidRipple === false ||
      (!this.props.onPress && !this.props.onLongPress) ||
      Platform.Version <= 21
    ) {
      return (
        <TouchableHighlight
          onPress={this.props.onPress}
          onLongPress={this.props.onLongPress}
          ref={c => (this._root = c)}
          underlayColor={variables.listBtnUnderlayColor}
          {...this.props}
          style={this.props.touchableHighlightStyle}
        >
          <View {...this.props}>{this.props.children}</View>
        </TouchableHighlight>
      );
    } else {
      return (
        <TouchableNativeFeedback
          ref={c => (this._root = c)}
          {...this.props}
        >
          <View style={{ marginLeft: -17, paddingLeft: 17 }}>
            <View {...this.props}>{this.props.children}</View>
          </View>
        </TouchableNativeFeedback>
      );
    }
  }
}

// ListItem.propTypes = {
//   ...TouchableHighlight.propTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ]),
//   touchableHighlightStyle: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.array
//   ]),
//   itemDivider: PropTypes.bool,
//   button: PropTypes.bool
// };

const StyledListItem = connectStyle(
  "NativeBase.ListItem",
  {},
  mapPropsToStyleNames
)(ListItem);

export { StyledListItem as ListItem };
