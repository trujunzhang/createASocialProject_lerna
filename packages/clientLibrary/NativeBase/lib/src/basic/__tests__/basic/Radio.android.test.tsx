import 'react-primitives'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Content } from '../../Content'
import { ListItem } from '../../ListItem'
import { Left } from '../../Left'
import { Right } from '../../Right'
import { Text } from '../../Text'
import { Radio } from '../../Radio'

// Note: test renderer must be required after react-native.

// jest.mock("Platform", () => {
//   const Platform = require.requireActual("Platform");
//   Platform.OS = "android";
//  return Platform;
// });

it('renders default radio', () => {
  const tree = renderer
    .create(
      <Content>
        <ListItem selected={true}>
          <Left>
            <Text>Lunch Break</Text>
          </Left>
          <Right>
            <Radio selected={true} />
          </Right>
        </ListItem>
        <ListItem selected={false}>
          <Left>
            <Text>Daily Stand Up</Text>
          </Left>
          <Right>
            <Radio selected={false} />
          </Right>
        </ListItem>
      </Content>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders custom radio', () => {
  const tree = renderer
    .create(
      <Content>
        <ListItem selected={true}>
          <Left>
            <Text>Lunch Break</Text>
          </Left>
          <Right>
            <Radio selected={true} color={'#f0ad4e'} selectedColor={'#5cb85c'} />
          </Right>
        </ListItem>
        <ListItem selected={false}>
          <Left>
            <Text>Daily Stand Up</Text>
          </Left>
          <Right>
            <Radio selected={false} color={'#f0ad4e'} selectedColor={'#5cb85c'} />
          </Right>
        </ListItem>
      </Content>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
