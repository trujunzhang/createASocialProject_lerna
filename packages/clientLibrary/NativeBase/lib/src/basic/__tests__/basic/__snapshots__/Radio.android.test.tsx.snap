// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom radio 1`] = `
<view
  name="keyboardAwareScrollView"
  style={Object {}}
>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Lunch Break
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="TouchableOpacity: "
        />
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Daily Stand Up
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="TouchableOpacity: "
        />
      </view>
    </view>
  </view>
</view>
`;

exports[`renders default radio 1`] = `
<view
  name="keyboardAwareScrollView"
  style={Object {}}
>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Lunch Break
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="TouchableOpacity: "
        />
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="left"
        style={Object {}}
      >
        <view
          name="View"
          style={Object {}}
        >
          <text
            style={Object {}}
          >
            Daily Stand Up
          </text>
        </view>
      </view>
      <view
        name="right"
        style={Object {}}
      >
        <view
          name="TouchableOpacity: "
        />
      </view>
    </view>
  </view>
</view>
`;
